composed-true-concidered-harmful.md

```diff
@@ -1,35 +1,583 @@
+
+## `dispatchEvent()` <a id="manual-dispatch"></a>
+
+So far we've mainly talked about the `click` event as automatically dispatched by the browser. There is, of course, a whole family of UA-dispatched UI events that can be addressed in the same manner (e.g. `animationend`/`copy`/`keydown`/`mouseover`/`paste`/`touch`, etc.). However, the real fun starts when you take that power into your own hands and start dispatching events on your own creation. For this, the browser supplies us with the [`dispatchEvent()`](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/dispatchEvent) method that hangs off of anything extended from `EventTarget`, which includes all of the `HTMLElement` based class of DOM elements. For this to do its magic we need to supply it an event to dispatch. We're given a number of events classes to creat our new event from (e.g. `new Event()`, `new MouseEvent()`, `new InputEvent()`, etc.), but for the most part working with `new Event(typeArg[, initDict])` only is more than enough. 
+
+Now, we're ready to dispatch an event.
+
+```js
+el.dispatchEvent('test-event');
+```
+
+Event dispatched!
+
+The event has a `type` of `test-event`, so a listener like the following will be able to hear it:
+
+```js
+el.addEventListener('test-event', (e) => console.log(e.type));
+// test-event
+```
+
+You can also listen for this event during the "capture" phase:
+
+```js
+const body = document.querySelector('body');
+body.addEventListener('test-event', (e) => console.log(e.type), true);
+// test-event
+```
+
+But, you won't be hearing it in the "bubble" phase:
+
+```js
+const body = document.querySelector('body');
+body.addEventListener('test-event', (e) => console.log(e.type));
+// ... ... Bueller?
+```
+
+This is because by default a `new Event()` (as well as all derivative event constructors) have `bubbles`, `cancelable`, and `composed` set to `false` by default. This is where the optional `initDict` argument of our event constructor comes into play. When you want to customize the values of these, you'll create your event like so:
+
+```js
+const event = new Event('test-event', {
+    bubbles: true,
+    cancelable: true,
+    composed: true,
+};
+```
+
+Or however best supports (or least harms? 😉) the use case in question. That means that if you only want your event to be available in the "capture" phase (which literally means it takes half the time for it to run synchronously through your application that it would if it were to also make a pass through the "bubble" phase) you can leave that out. Don't have an action that you'd like permission to do? You can leave out `cancelable`, too. Don't have shadow DOM? Decided definatively that `composed: true` is harmful? It's your rodeo, leave it out!
```

Comment:
```suggestion
Or however best supports (or least harms? 😉) the use case in question. That means that if you only want your event to be available in the "capture" phase (which literally means it takes half the time for it to run synchronously through your application than it would if it were to also make a pass through the "bubble" phase) you can leave that out. Don't have an action that you'd like permission to do? You can leave out `cancelable`, too. Don't have shadow DOM? Decided definitely that `composed: true` is harmful? It's your rodeo, leave it out!
```